src/main/java/com/incentives/piggyback/offers/serviceImpl/OfferServiceImpl.java

```diff
@@ -1,58 +1,92 @@
-            //PUSHING MESSAGES TO GCP
-            messagingGateway.sendToPubsub(
-                    CommonUtility.stringifyEventForPublish(
-                            offers.toString(),
-                            Constant.OFFER_CREATED_EVENT,
-                            Calendar.getInstance().getTime().toString(),
-                            "",
-                            Constant.OFFER_SOURCE_ID
-                    ));
-            return ResponseEntity.ok(offers);
-    }
-
-    private OfferDTO updateOffer(OfferDTO offer) {
-       // as these are mandatory fields and should be present for update
-        if(offer.getOfferId()!=null && offer.getPartnerId()!=null && offer.getOfferCode()!=null){
-
-                if (OfferStatus.getAllStatus().contains(offer.getOfferStatus()))
-                    offer.setOfferStatus(offer.getOfferStatus());
-                else
-                    throw new InvalidRequestException("Invalid Status");
-
-                //date need to be updated to activate the offer
-                if(null!=offer.getOfferValidTill())
-                    offer.setOfferValidTill(offer.getOfferValidTill());
-                else
-                    throw new InvalidRequestException("Offer Validity date is not passed");
-
-            return offer;
-        }else
-        {
-            throw new InvalidRequestException("Offer Id or Order Id or OfferCode cannot be null");
-        }
-    }
+	@Autowired
+	private OffersEventPublisher.PubsubOutboundGateway messagingGateway;
+
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Override
+	public ResponseEntity<OfferDTO> updateOfferStatus(Long id, OfferDTO offers)  {
+		updateOffer(offers);
+		//PUSHING MESSAGES TO GCP
+		messagingGateway.sendToPubsub(
+				CommonUtility.stringifyEventForPublish(
+						offers.toString(),
+						Constant.OFFER_CREATED_EVENT,
+						Calendar.getInstance().getTime().toString(),
+						"",
+						Constant.OFFER_SOURCE_ID
+						));
+		return ResponseEntity.ok(offers);
+	}
+
+	private OfferDTO updateOffer(OfferDTO offer) {
+		// as these are mandatory fields and should be present for update
+		if(offer.getOfferId()!=null && offer.getPartnerId()!=null && offer.getOfferCode()!=null){
+
+			if (OfferStatus.getAllStatus().contains(offer.getOfferStatus()))
+				offer.setOfferStatus(offer.getOfferStatus());
+			else
+				throw new InvalidRequestException("Invalid Status");
+
+			//date need to be updated to activate the offer
+			if(null!=offer.getOfferValidTill())
+				offer.setOfferValidTill(offer.getOfferValidTill());
+			else
+				throw new InvalidRequestException("Offer Validity date is not passed");
+
+			return offer;
+		} else {
+			throw new InvalidRequestException("Offer Id or Order Id or OfferCode cannot be null");
+		}
+	}
+
+	@Override
+	public List<LocationDTO> getNearbyUsers(Long userId, Double latitude, 
+			Double longitude, Integer page) {
+		String url = "http://localhost:8080/location/user";
```

Comment:
localhost url should be changed to container related url

Reply:
This should be fine for now. Even if it is pointed to localhost, the docker-compose deployment is on a single machine and will still work.
We will need to change this once services are deployed on Kubernetes.